pilot/pkg/bootstrap/server.go

```diff
@@ -220,6 +222,31 @@ func NewServer(args *PilotArgs) (*Server, error) {
 		return nil, fmt.Errorf("error initializing kube client: %v", err)
 	}
 
+	if features.EnableSDSServer && s.kubeClient != nil {
+		if !features.EnableXDSIdentityCheck {
+			// Make sure we have security
+			log.Warnf("skipping Kubernetes credential reader. PILOT_ENABLE_XDS_IDENTITY_CHECK must be set to true for this feature.")
```

Comment:
might be better to call out this is because of the EnableSDSServer flag, so that users don't need to connect the "Kubernetes credential reader" to EnableSDSServer, which is not super obvious from the name.